common/src/main/java/io/crate/interval/IntervalParser.java

```diff
@@ -0,0 +1,125 @@
+        return apply(value, null, null);
+    }
+
+    public static Period apply(String value, Precision start, Precision end) {
+        Period result;
+        try {
+            result = NumericalIntervalParser.apply(value, start, end);
+        } catch (IllegalArgumentException e1) {
+            try {
+                result = IsoIntervalParser.apply(value, start, end);
+            } catch (IllegalArgumentException e2) {
+                try {
+                    result = SQLStandardIntervalParser.apply(value, start, end);
+                } catch (IllegalArgumentException e3) {
+                    result = PGIntervalParser.apply(value, start, end);
+                }
+            }
+        }
+        return result.normalizedStandard();
+    }
+
+    static Period roundToPrecision(Period period, Precision start, Precision end) {
+        if (start == null && end == null) {
+            return period;
+        }
+        if (start == Precision.YEAR && end == null) {
+            return Period.years(period.getYears());
+        }
+        if (start == Precision.YEAR && end == Precision.MONTH) {
+            return Period.years(period.getYears()).withMonths(period.getMonths());
+        }
+        if (start == Precision.MONTH && end == null) {
+            return Period.years(period.getYears()).withMonths(period.getMonths());
+        }
+        if (start == Precision.DAY && end == null) {
+            return new Period(period).withHours(0).withMinutes(0).withSeconds(0).withMillis(0);
+        }
+        if (start == Precision.DAY && end == Precision.HOUR) {
+            return new Period(period).withMinutes(0).withSeconds(0).withMillis(0);
+        }
+        if (start == Precision.DAY && end == Precision.MINUTE) {
+            return new Period(period).withSeconds(0).withMillis(0);
+        }
+        if (start == Precision.DAY && end == Precision.SECOND) {
+            return new Period(period).withMillis(0);
+        }
+        if (start == Precision.HOUR && end == null) {
+            return new Period(period).withMinutes(0).withSeconds(0).withMillis(0);
+        }
+        if (start == Precision.HOUR && end == Precision.MINUTE) {
+            return new Period(period).withSeconds(0).withMillis(0);
+        }
+        if (start == Precision.HOUR && end == Precision.SECOND) {
+            return new Period(period).withMillis(0);
+        }
+        if (start == Precision.MINUTE && end == null) {
+            return new Period(period).withSeconds(0).withMillis(0);
+        }
+        if (start == Precision.MINUTE && end == Precision.SECOND) {
+            return new Period(period).withMillis(0);
+        }
+        if (start == Precision.SECOND && end == null) {
+            return new Period(period).withMillis(0);
+        }
+        throw new IllegalArgumentException(String.format("Invalid start and end combination", start, end));
+    }
+
+    static int parseMilliSeconds(String value) throws NumberFormatException {
```

Comment:
I think this could use some doc to explain the format `value` is supposed to be in.